nosetimer/plugin.py

```diff
@@ -149,10 +151,13 @@ def report(self, stream):
         d = sorted(self._timed_tests.items(),
                    key=lambda item: item[1]['time'],
                    reverse=True)
+
         if self.json_file:
-            import json
             with open(self.json_file, 'w') as f:
-                json.dump({'tests': dict((k, v) for k, v in d)}, f)
+                if self.timer_top_n:
+                    json.dump({'tests': OrderedDict((k, v) for k, v in d)}, f)
+                else:
+                    json.dump({'tests': dict((k, v) for k, v in d)}, f)
```

Comment:
Done.

Remove the duplication but put it into a separate step. Readability matters. :)